### PR TITLE
Expose beam search diversity rate for ft

### PR DESCRIPTION
**Pull request by FrostML**
### PR types

New features

### PR changes

APIs

### Description

* Expose beam search diversity rate for ft. 
* Fix #1041, which change the interface of generate api. 
* Fix BART dtype exception.

### CODE REVIEW

File: examples/machine_translation/transformer/faster_transformer/README.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ python encoder_decoding_predict.py --config ../configs/transformer.base.yaml --d
   * 当使用 `topk_sampling` 的时候，需要指定 `--topk` 的值
   * 当使用 `topp_sampling` 的时候，需要指定 `topp` 的值，并且需要保证 `--topk` 的值为 0
 * `--beam_size`: 解码策略是 `beam_search` 的时候，beam size 的大小，数据类型是 `int`
+* `--beam_search_diversity_rate`: 解码策略是 `beam_search` 的时候，设置 diversity rate 的大小，数据类型是 `float`。当设置的 `beam_search_diversity_rate` 大于 0 的时候，FasterTransformer 仅支持 beam size 为 1，4，16，64
```
Comment (smallv0221) [inline examples/machine_translation/transformer/faster_transformer/README.md]: 这里暴露出来的参数名字跟生成api统一吧，都叫diversity_rate

**Review by smallv0221 — COMMENTED** [review]
需不需要支持一下任意beam_size呢

File: examples/machine_translation/transformer/faster_transformer/README.md (hunk 1 above)
Comment (FrostML) [reply examples/machine_translation/transformer/faster_transformer/README.md]: Done. Thanks.
已修改 Transformer，BART，UnifiedTransformer。验证中。

**Review by smallv0221 — DISMISSED** [review]
LGTM

**Review by gongel — DISMISSED** [review]
LGTM
测试没问题就行😊

**Review by smallv0221 — APPROVED** [review]
LGTM

**Review by gongel — APPROVED** [review]

**Pull request merged**